tasks/posternak_a_count_different_char_in_two_lines/report.md

```diff
@@ -0,0 +1,351 @@
+- `RunImpl()` - реализация SEQ/MPI алгоритма
+- `PostProcessingImpl()` - завершающая обработка
+
+**Частные случаи:**
+Предварительное условие - строки должны быть непустыми
+
+## 6. Экспериментальное окружение
+
+**Аппаратное обеспечение:**
+- Процессор: Intel Core i7-11800H @ 2.30GHz
+- Ядра: 16 шт.
+- ОЗУ: 16 ГБ
+- ОС: Kubuntu 25.10
+
+**Программный инструментарий:**
+- Компилятор: g++ 15.2.0
+- Тип сборки: Release
+- Стандарт C++: C++23
+- MPI: Open MPI 5.0.8
+
+**Тестовое окружение**
+```bash
+PPC_NUM_PROC=1,2,4
+```
+
+## 7. Результаты
+
+### 7.1 Корректность
+
+Все функциональные тесты были успешно пройдены:
+1. Строки одинаковой длины с различием в 1 символ ("string" и "strong").
+2. Длина первой строки больше длины второй ("stringstring" и "string").
+3. Длина второй строки больше длины первой ("string" и "stringstring").
+4. Одиннаковые строки ("string" и "string"). 
+5. Строки с разным регистром ("string" и "STRING").
+
+SEQ и MPI версии выдают идентичные результаты для всех тестовых случаев.
+
+### 7.2 Производительность
+
+**Результаты замера времени выполнения MPI и SEQ алгоритмов для строк, длинной 10.000.000 символов:**
+
+| Режим | Количество процессов | Время, с | Ускорение | Эффективность |
+|-------|----------------------|----------|-----------|---------------|
+| SEQ   | 1                    | 17.14    | 1.00      | N/A           |
+| MPI   | 1                    | 17.66    | 0.97      | 97%           |
+| MPI   | 2                    | 11.00    | 1.56      | 78%           |
+| MPI   | 4                    | 7.72     | 2.22      | 56%           |
+
+
+**Формула ускорения:** Ускорение = Время SEQ / Время MPI
+
+**Формула эффективности:** Эффективность = (Ускорение / Количество процессов) × 100%
+
+### 7.3. Анализ результатов
+
+**Лучшее ускорение:** 2.22 на 4 процессах
+
+**Эффективность:** При увеличении числа процессов снижается показатель эффективности
+
+## 8. Выводы
+
+В результате выполнения проекта были разработаны и протестированы последовательный (SEQ) и параллельный (MPI) алгоритма подсчета различающихся символов в двух строках.
+
+Параллельный алгоритм работает быстрее последовательного на больших данных, однако на малых немного уступает по времени выполнения от последовательного алгоритма из-за обмена данными между процессами.
+
+При увеличении числа ядер для параллельного алгоритма уменьшается эффективность. В основном, это связанно с увеличением объема обмена данными между процессами.
+
+Итого, для малых объемов данных лучше использовать последовательный (SEQ) алгоритм, а для больших объемов данных - параллельный (MPI) алгоритм.
+
+## 9. Литература
+
+1. Документация по курсу: "Параллельное программирование": https://learning-process.github.io/parallel_programming_course/ru/index.html (Оболенский А.А, Нестеров А.Ю)
+2. Лекции по курсу "Параллельное программирование". (Сысоев А.В. ННГУ 2025 г.)
+3. Документация по MPI: https://www.open-mpi.org/
+
+## Приложение
```

Comment:
If the appendix is just a copy-paste of implementation files, I'd rather remove it